lib/logdna/client.rb

```diff
@@ -125,14 +123,20 @@ def flush
           send_request
         end
         @lock.unlock
+        Thread.current.exit
       else
         schedule_flush
       end
     end
 
     def exitout
-      flush
-      @internal_logger.debug("Exiting LogDNA logger: Logging remaining messages")
+      if @buffer.any? || @side_messages.any?
+        @internal_logger.debug("Exiting LogDNA logger: Logging remaining messages")
+        Concurrent::Future.execute({ executor: @thread_pool }) do
+          flush
+        end
+        @internal_logger.debug("Finished flushing logs to LogDNA")
```

Comment:
`Concurrent::Future.execute()` is non-blocking, so the "finished" log message should be after the flush, right? also do we need to call `@thread_pool.shutdown()` here?

finally, I wonder if in `exitout` this should be run in a thread at all. doing so means that the logger client `close()` is going to return before the flush is complete. I'm not sure if that's an issue or not, as long as the thread will complete before the application terminates.

Reply:
yep agreed...  fixing this